File: Serene3/Serene3.Web/Modules/ServerTypes/com/review2Form.ts
import { DateEditor, initFormType, IntegerEditor, PrefixedContext, ServiceLookupEditor, StringEditor } from "@serenity-is/corelib";

export interface review2Form {
    ProductId: ServiceLookupEditor;
    Name: StringEditor;
    Email: StringEditor;
    Subject: StringEditor;
    Description: StringEditor;
    Rating: IntegerEditor;
    CreatedAt: DateEditor;
}

export class review2Form extends PrefixedContext {
    static readonly formKey = 'com.review2';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!review2Form.init) {
            review2Form.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = StringEditor;
            var w2 = IntegerEditor;
            var w3 = DateEditor;

            initFormType(review2Form, [
                'ProductId', w0,
                'Name', w1,
                'Email', w1,
                'Subject', w1,
                'Description', w1,
                'Rating', w2,
                'CreatedAt', w3
            ]);
        }
    }
}

File: Serene3/Serene3.Web/Modules/com/review2/review2Form.cs
using Serenity.ComponentModel;
using System;

namespace Serene3.com.Forms;

[FormScript("com.review2")]
[BasedOnRow(typeof(review2Row), CheckNames = true)]
public class review2Form
{
    public int ProductId { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public string Subject { get; set; }
    public string Description { get; set; }
    public int Rating { get; set; }
    public DateTime CreatedAt { get; set; }
}